File: backend/src/main/java/com/odontosystem/api/entity/Appointment.java
package com.odontosystem.api.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

/**
 * Entidad Cita. Relaciona un paciente (User) con un odontólogo (Dentist).
 */
@Entity
@Table(name = "appointments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Appointment {

    @Id
    @GeneratedValue
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "patient_id", nullable = false)
    private User patient;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "dentist_id", nullable = false)
    private Dentist dentist;

    @Column(nullable = false)
    private LocalDate date;

    @Column(nullable = false, length = 10)
    private String time;

    @Column(length = 500)
    private String reason;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private AppointmentStatus status = AppointmentStatus.Pendiente;

    @Column(name = "created_at", nullable = false, updatable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: app/src/main/java/com/odontosystem/app/data/model/Appointment.kt
package com.odontosystem.app.data.model

data class Appointment(
    val id: String,
    val dentistId: String,
    val dentistName: String,
    val specialty: String,
    val district: String,
    val date: String,
    val time: String,
    val reason: String?,
    val status: String = "Pendiente" // Backend: "Pendiente", "Confirmada", "Cancelada" (RF-04: + "Completada")
)

data class CreateAppointmentRequest(
    val dentistId: String,
    val date: String,
    val time: String,
    val reason: String?
)
